mirrord/auth/src/credentials.rs

```diff
@@ -13,60 +13,53 @@ use crate::{
     key_pair::KeyPair,
 };
 
-/// Client Credentials container for authorization against Operator
+/// Client credentials container for authentication with the operator.
+/// Contains a local [`KeyPair`] and an optional [`Certificate`].
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Credentials {
-    /// Generated Certificate from operator can be None when request isn't signed yet and will
-    /// result in is_ready returning false
-    certificate: Option<Certificate>,
-    /// Local Keypair for creating CertificateRequest for operator
+    /// Certificate generated by the operator based on the sent [`rfc2986::CertificationRequest`].
+    certificate: Certificate,
```

Comment:
Optional `Certificate`?

Reply:
It used to be an `Option` due to code logic (the flow included creating "dummy" `Credentials`), but it was never saved to the file before inserting the `Certificate`. I changed the code so that it doesn't have to be an `Option` now